### PR TITLE
Docker build

### PR DESCRIPTION
**Pull request by palp**
This change adds a Dockerfile to make a consistent build environment easily available.

### CODE REVIEW

**Comment by salanki** [comment]
Will this build for both python and js?

**Comment by palp** [comment]
> Will this build for both python and js?

Yeah, and thinking on it, I'd like to exclude those files from the build to ensure they get freshly compiled each run, since the output is checked in - that'd make it easily testable too.

**Comment by palp** [comment]
One notable tradeoff here is that I decided on compiling protobuf and grpc rather than relying on binaries/packages - with cached images this seems fine and I think that's the best way to ensure we can have the latest versions of those cleanly.

**Comment by palp** [comment]
I've added a workflow to build using this image that uploads the output as an artifact; my thinking is we could update #15 to depend on this and remove the build steps, keeping the check-in code, as well as publish the artifacts as a release - all scoped to main. This one can run on push (or PR and push to main) so there's a build check without publish.

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+RUN mkdir -p /build
+
+# Build GRPC; note this issue before updating: https://github.com/protocolbuffers/protobuf-javascript/issues/127
+WORKDIR /build
+RUN git clone --recurse-submodules -b v1.48.2 --depth 1 --shallow-submodules https://github.com/grpc/grpc
+WORKDIR /build/grpc
+RUN mkdir -p cmake/build; cd cmake/build; cmake -DgRPC_INSTALL=ON -DgRPC_BUILD_TESTS=OFF -DCMAKE_INSTALL_PREFIX=/usr/local ../.. && make -j 8 && make install
+
+# Build api-interfaces
+RUN go install google.golang.org/grpc/cmd/protoc-gen-go-grpc@v1.1.0
```
Comment (wbrown) [inline Dockerfile:18]: Can we pull these versions up to variables?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,27 @@
+FROM node:16-bullseye as builder
+RUN apt-get update && apt-get install -y cmake git build-essential python3-venv wget && rm -rf /var/lib/apt/lists/*
+
+# Install golang 
+RUN rm -rf /go && wget https://go.dev/dl/go1.19.1.linux-amd64.tar.gz -qO- | tar -C / -xz
```
Comment (wbrown) [inline Dockerfile:5]: I think we're still on go1.18, and if using go1.19 imposes a dependency on that with the artifacts, it may be problematic.

**Review by wbrown — CHANGES_REQUESTED** [review]

**Review by wbrown — APPROVED** [review]

**Pull request merged**